buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidLibTarget.groovy

```diff
@@ -22,4 +26,23 @@ class AndroidLibTarget extends AndroidTarget {
     ManifestMerger2.MergeType getMergeType() {
         return ManifestMerger2.MergeType.LIBRARY
     }
+
+    List<String> getKotlincArguments() {
+        ImmutableList.Builder<String> extraKotlincArgs = ImmutableList.<String>builder()
+        if (hasKotlinAndroidExtensions) {
+            String options = ""
+            getResDirs().each {  String dir ->
+                // :root:module -> /root/module/
+                String module = project.path.replace(":", "/") + "/"
+                // src/main/res -> main
+                String variant = dir.substring(dir.indexOf("/") + 1, dir.lastIndexOf("/"))
+                String pathToRes = project.rootDir.toPath().toAbsolutePath().toString() + module + dir
```

Comment:
Path to res is already available via the model which is how `getResDirs()` works. We should use the data from the model directly instead of convoluting it like this

Reply:
I need the full absolute path though, is that available and I'm just missing something? If I remember correctly I needed to construct my own path especially for the module name because we have some that are 2 levels deep and that wasn't being accounted for since the project name was only the last of the two directories.

Reply:
I need the full absolute path though, is that available and I'm just missing something? If I remember correctly I needed to construct my own path especially for the module name because we have some that are 2 levels deep and that wasn't being accounted for since the project name was only the last of the two directories.

Reply:
We have to use relative paths or the buck network cache will be rendered ineffective